### ISSUE TITLE
Issue of generating reciprocal edges in directed graph

### ISSUE THREAD

**Issue by c752334430** [JiaxuanYou/graph-generation#14]
Hi, thanks for open source the code! The framework can learn and generate undirected graph with high quality. But when I train the model (modified based on the suggestion given in the appendix of the paper) with directed graph, I find it cannot generate similar amount of reciprocal edges compare to the training graph I have.

Specifically, The modification I have done is double the length of edge sequence of node i as (A_1i, A_i1, A_2i, A_i2 ...), where A is the adj matrix. I do know this leads to a sparser sequence.

The graph I'm training has around 200 nodes, 659 one sided directed edges and 58 reciprocal edges, while the generated graph in average has less than 520 directed edges and 15 to 20 reciprocal edges. So the model is generating sparser graph.

I wonder if anyone has experience using this framework with directed graph, and give any advise on dealing with my issue?

Thanks in advance.

**Comment by ZeroCSIS** [comment]
Hello, I have some similar questions, may I have a discussion with you? It seems that you are also working on graphs.

**Comment by c752334430** [comment]
Sure, send an email to ctianyi7@gmail.com so we can schedule a chat :D

**Comment by thesaturdayafternoon** [comment]
Hello,
I just saw your post and I wonder if you have succeeded to use graphRNN code to generate graphs with costume node labels or with node attributes? By modifying 'create_graphs.py', I created a training set of grid2D graphs where instead of default labels such as (0,0), (0,1), (1,0),...nodes have costume node labels such as a,b,c,... and they have also a node attribute. However when I train graphRNN with these graphs, then the  generated graphs have default labels (0,0), (0,1), (1,0),...and no node attributes. Do you have any solution for this problem?
Thanks
